### PR TITLE
[SPARK-46245][CORE][SQL][SS][YARN][K8S][UI] Replcace `s.c.MapOps.view.filterKeys` with `s.c.MapOps.filter`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr uses `s.c.MapOps.filter` to simplify code pattern `s.c.MapOps.view.filterKeys`.

### Why are the changes needed?
The coding pattern of `s.c.MapOps.view.filterKeys` seems verbose, it can be simplified using `s.c.MapOps.filter`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
Just a question.
- Are we going to remove all `.view.`?
- How many PRs do we need?

**Comment by LuciferYang** [comment]
> Are we going to remove all .view.?

Yes, for ·MapOps.view·, this last pr, the reason for making this pr can be referred to https://github.com/apache/spark/pull/43445#discussion_r1413338062. 

> How many PRs do we need?

As for the remaining `Seq.view` and `Array.view` in the spark code, they were not introduced by the recent refactoring work, 
I am investigating their necessity now. If cleanup is needed, it will be completed in one.

File: core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala [hunk 1]
```diff
@@ -254,8 +254,8 @@ private[spark] class StandaloneSchedulerBackend(
 
   override def getDriverLogUrls: Option[Map[String, String]] = {
     val prefix = "SPARK_DRIVER_LOG_URL_"
-    val driverLogUrls = sys.env.view.filterKeys(_.startsWith(prefix))
-      .map(e => (e._1.substring(prefix.length).toLowerCase(Locale.ROOT), e._2)).toMap
+    val driverLogUrls = sys.env.filter { case (k, _) => k.startsWith(prefix) }
```
[the remark this answers is not in the document]
Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/scheduler/cluster/StandaloneSchedulerBackend.scala:257]: I don't have a strong opinion. Both are fine.

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for the answers and making progress in this area, @LuciferYang .

**Comment by dongjoon-hyun** [comment]
All tests passed. Merged to master for Apache Spark 4.0.0.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun and @cloud-fan

**Pull request closed without merging**